certbot/auth_handler.py

```diff
@@ -67,9 +67,9 @@ def handle_authorizations(self, orderr, best_effort=False):
         config = zope.component.getUtility(interfaces.IConfig)
         notify = zope.component.getUtility(interfaces.IDisplay).notification
 
-        with error_handler.ExitHandler(self._cleanup_challenges, aauthzrs):
-            # While there are still challenges remaining...
-            while self._has_challenges(aauthzrs):
+        # While there are still challenges remaining...
+        while self._has_challenges(aauthzrs):
+            with error_handler.ExitHandler(self._cleanup_challenges, aauthzrs):
```

Comment:
Cleaned up remaining `_cleanup_challenges` call-- good catch! As a result, had to swap the ordering here (so that each successful auth is cleaned up individually).

Reply:
Nice catch!